### PR TITLE
Improving test coverage and adding coverage reporting script

### PR DESCRIPTION
**Pull request by rodrigobr-msft**
This pull request introduces several improvements and fixes across the codebase, primarily focused on the `microsoft-agents-activity` library. The main themes are enhanced test coverage, improvements to type handling and serialization, and minor refactoring for clarity and maintainability. Key changes include the addition of comprehensive unit tests, improvements to type annotations and field aliasing, and updates to activity type handling logic.

**Test coverage and validation improvements:**

* Added extensive unit tests for `Entity`, `ConversationReference`, and model utility functions to ensure correct validation, serialization, and field processing behavior. This includes tests for required fields, alias handling, and serialization edge cases (`tests/activity/entity/test_entity.py`, `tests/activity/test_conversation_reference.py`, `tests/activity/test_model_utils.py`). [[1]](diffhunk://#diff-52845153e106f29be1ba0b6a65fb8d641fa8c7e169e33df531830368aeaab0eeR1-R82) [[2]](diffhunk://#diff-a3461cfe27b40bc17768557821b24401c0f696efc9cbeb7d387d88032b00a6e0R1-R58) [[3]](diffhunk://#diff-d969efdfad06dd52bdcd37b35dfc5fb844e16fe5080eabad7b39da9707085c06R1-R164)
* Enhanced tests for `Activity` to cover type helper methods and agentic tenant ID retrieval logic, increasing confidence in correct type discrimination and role handling (`tests/activity/test_activity.py`). [[1]](diffhunk://#diff-7ced027634467957be20d27cda02e88c6ec4f947a42aa889b410ad09e0510c7fR29-R45) [[2]](diffhunk://#diff-7ced027634467957be20d27cda02e88c6ec4f947a42aa889b410ad09e0510c7fR259-R260) [[3]](diffhunk://#diff-7ced027634467957be20d27cda02e88c6ec4f947a42aa889b410ad09e0510c7fR474-R507) [[4]](diffhunk://#diff-7ced027634467957be20d27cda02e88c6ec4f947a42aa889b410ad09e0510c7fR577-R629)

**Type annotations and field aliasing:**

* Updated field definitions in `ConversationReference` and `TokenExchangeState` to use `Annotated` for improved type clarity and explicit aliasing, aligning with modern Pydantic practices (`microsoft_agents/activity/conversation_reference.py`, `microsoft_agents/activity/token_exchange_state.py`). [[1]](diffhunk://#diff-0c47acd7d95ec916e7895ba99f7b13a3318cc6f120a0725dfb7664ba24525616L7-L22) [[2]](diffhunk://#diff-0c47acd7d95ec916e7895ba99f7b13a3318cc6f120a0725dfb7664ba24525616L51-R47) [[3]](diffhunk://#diff-3492941655b9cd8bae209067e8040d9e4aeb8c56b9ec545f6051ad4743f4fb1aL6-R6) [[4]](diffhunk://#diff-3492941655b9cd8bae209067e8040d9e4aeb8c56b9ec545f6051ad4743f4fb1aL32-R32)

**Activity type handling logic:**

* Refactored the internal activity type comparison logic in `Activity` to more robustly handle both enum and string representations and to support slash-qualified activity types (`activity.py`).

**Test and coverage tooling:**

* Added a `.coveragerc` file to omit testing package modules from coverage reports and updated the coverage script to generate HTML reports (`.coveragerc`, `scripts/coverage.ps1`). [[1]](diffhunk://#diff-834e9b406d74791ffbafaeec9cc894082cea9739bc347b6ff9f72312c92ebc79R1-R3) [[2]](diffhunk://#diff-8de5118ef7a82ac01873d836cf3d048332fc3c9be0c317b6afd94416b6f2eb77R1-R2)

**Minor refactoring and cleanup:**

* Removed unused imports and logging setup from `conversation_reference.py` for clarity (`microsoft_agents/activity/conversation_reference.py`).

These changes collectively improve code quality, maintainability, and confidence in correctness through better testing and type handling.

### CODE REVIEW

File: libraries/microsoft-agents-activity/microsoft_agents/activity/conversation_reference.py [hunk 1]
```diff
@@ -48,7 +44,7 @@ class ConversationReference(AgentsModel, _ChannelIdFieldMixin):
     # optionals here are due to webchat
     activity_id: Optional[NonEmptyString] = None
     user: Optional[ChannelAccount] = None
-    agent: ChannelAccount = Field(None, alias="bot")
+    agent: Annotated[ChannelAccount, Field(None, alias="bot")] = None
```
[the remark this answers is not in the document]
Comment (rodrigobr-msft) [reply libraries/microsoft-agents-activity/microsoft_agents/activity/conversation_reference.py]: This is a convention we use. Not a problem.

File: .coveragerc [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[run]
+omit =
+   */microsoft-agents-testing/*
```
[the remark this answers is not in the document]
Comment (rodrigobr-msft) [reply .coveragerc:3]: I will update the PR desc.

File: libraries/microsoft-agents-activity/microsoft_agents/activity/token_exchange_state.py [hunk 3]
```diff
@@ -29,7 +29,7 @@ class TokenExchangeState(AgentsModel):
     connection_name: NonEmptyString = None
     conversation: ConversationReference = None
     relates_to: Optional[ConversationReference] = None
-    agent_url: NonEmptyString = Field(None, alias="bot_url")
+    agent_url: Annotated[NonEmptyString, Field(alias="bot_url")] = None
```
[the remark this answers is not in the document]
Comment (rodrigobr-msft) [reply libraries/microsoft-agents-activity/microsoft_agents/activity/token_exchange_state.py:32]: Again, convention.

File: libraries/microsoft-agents-hosting-core/microsoft_agents/hosting/core/app/state/temp_state.py [hunk 4]
```diff
@@ -64,9 +64,13 @@ def delete_value(self, name: str) -> None:
...
     def get_value(
-        self, name: str, default_value_factory: Optional[Callable[[], T]] = None
-    ) -> T:
-        """Gets a value from the state with the given name, using a factory for default values if not found"""
+        self,
+        name: str,
+        default_value_factory: Optional[Callable[[], T]] = None,
+        *,
+        target_cls: type[T] | None = None,
+    ) -> T | None:
```
[the remark this answers is not in the document]
Comment (rodrigobr-msft) [reply libraries/microsoft-agents-hosting-core/microsoft_agents/hosting/core/app/state/temp_state.py:72]: That is okay because TempState stores objects in memory, so no serialization/deserialization is needed

**Review by kylerohn-msft — APPROVED** [review]
